pkg/provider/bitbucketserver/parse_payload.go

```diff
@@ -155,7 +155,6 @@ func (v *Provider) ParsePayload(_ context.Context, _ *params.Run, request *http.
 		}
 		v.pullRequestNumber = e.PullRequest.ID
 	case *types.PushRequestEvent:
-		processedEvent.Event = "push"
```

Comment:
can you explain me why this is fixing it ?

we do the same on for example gitlab https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/provider/gitlab/parse_payload.go#L112-L112 does that change anything ?

Reply:
@chmouel Yes, but if you see it is [TriggerTarget](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/params/info/events.go#L26) not [Event](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/params/info/events.go#L11) in [gitlab's parse_payload](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/provider/gitlab/parse_payload.go#L112-L112) as well.

Reply:
@chmouel I think that [processedEvent.Event](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/params/info/events.go#L11) is not meant for trigger type (string) it is for storing raw payload of event, so the error in #1677 is `"there was an error evaluating the CEL expression, skipping: json: cannot unmarshal string into Go value of type map[string]interface {}"`. which was happening [here](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/matcher/cel.go#L54) in code. First [event.Event](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/params/info/events.go#L11) is [marshaled](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/matcher/cel.go#L49) and then unmarshaled and it was [set](https://github.com/openshift-pipelines/pipelines-as-code/blob/f7c282bbdc8776abe7a39352c8aa4af9b8177a6e/pkg/provider/bitbucketserver/parse_payload.go#L158) to `"push"` string (which I deleted in this PR) there was an error while unmarshaling it because it was just a string `"push"` and thus cel wasn't proceeding further and was giving an error when event is push in CEL expression and it works for pull_request finely.

Reply:
ah yes you right, that was a mistake, i misread